app/assets/javascripts/events/edit_event_form.js

```diff
@@ -0,0 +1,35 @@
+var editEventForm = {
+    timezone : function() {
+        return timeZoneUtilities.detectUserTimeZone();
+    },
+    handleUserTimeZone: function () {
+        if ($("#start_time_tz").length) {
+            $('#start_time_tz').setToUserTimeZone(this.timezone());
+        }
+        if ($('#start_time').length && $('#start_date').length) {
+            this.updateDateAndTimeToUserTimeZone($('#next_date'), $('#start_date'), $('#start_time'));
+        }
+    },
+    updateDateAndTimeToUserTimeZone : function(next_date, start_date, start_time){
+        var normalized_next_date_time = moment.utc(next_date.val() + " " + start_time.val(), "YYYY-MM-DD hh:mm a");
+        var normalized_start_date_time = moment.utc(start_date.val() + " " + start_time.val(), "YYYY-MM-DD hh:mm a");
```

Comment:
may be it's possible to create some layer of indirection and elimate explicit dependency on **moment** and have some object that encapsulate all the functionality related to timezone or time on general